### PR TITLE
DBZ-3133 Smoke and Unit testing integration

### PR DESCRIPTION
**Pull request by uidoyen**
Implement GitHub service containers to spin up test database/s, `zookeeper`, `kafka`, `postgres` and `connect`
Build UI inside GH Action
Build back-end that includes the UI as well  and run back-end container inside GitHub Action
Running Cypress tests inside GH Action on push which:
- Makes sure the Debezium UI starts up on `8080` port
- Can connect to Kafka Connect 
- Able to connect back-end and get connectors list
- Can navigate to create connector wizard step

Implement Unit test using `Jest` and `Enzyme` with code coverage
Running Unit test inside GitHub Actions on push

### CODE REVIEW

**Comment by mdrillin** [comment]
Hey @uidoyen - are the following files generated?  (should they be excluded and put into .gitignore?)
/ui/packages/ui/coverage/clover.xml
/ui/packages/ui/coverage/coverage-final.json
/ui/packages/ui/coverage/lcov-report/PageLoader.tsx.html
/ui/packages/ui/coverage/lcov-report/index.html

**Comment by uidoyen** [comment]
> Hey @uidoyen - are the following files generated? (should they be excluded and put into .gitignore?)
> /ui/packages/ui/coverage/clover.xml
> /ui/packages/ui/coverage/coverage-final.json
> /ui/packages/ui/coverage/lcov-report/PageLoader.tsx.html
> /ui/packages/ui/coverage/lcov-report/index.html

Hey @mdrillin Yes, these files are generated, adding into `.gitignore` to exclude

**Comment by rk3rn3r** [comment]
@uidoyen pls also add to `.dockerignore`  where applicable

**Comment by uidoyen** [comment]
> @uidoyen pls also add to `.dockerignore` where applicable
@rk3rn3r Updated  `.dockerignore` file

**Review by indraraj — APPROVED** [review]
tested locally smoke test and unit test setup seems to be working fine

**Review by mdrillin — APPROVED** [review]
Thanks for updating!

File: .dockerignore [hunk 1]
```diff
@@ -2,4 +2,4 @@
 .mvn/
 mvnw*
 .github/
-oc-demo/
+oc-demo/
```
Comment (rk3rn3r) [inline .dockerignore:5]: What happened here? :thinking:

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          GROUP_ID: 1
+          CONFIG_STORAGE_TOPIC: my_connect_configs
+          OFFSET_STORAGE_TOPIC: my_connect_offsets
+          STATUS_STORAGE_TOPIC: my_connect_statuses   
+
+    steps:
+      - name: Checkout Debezium UI
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 11
```
Comment (rk3rn3r) [inline .github/workflows/integration.yml:42]: Do you use Java at all? Maybe this can be removed?

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Cache local Maven repository
```
Comment (rk3rn3r) [inline .github/workflows/integration.yml:47]: Same here. When you don't use Maven, you don't need to cache its repo.
Instead we should file an issue about adding caching of the docker build cache and intermediate images.

File: .github/workflows/integration.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: dbzui-${{ runner.os }}-maven-${{ hashFiles('pom.xml', '**/pom.xml') }}
+          restore-keys: |
+            dbzui-${{ runner.os }}-maven-
+
+      - name: Get yarn cache directory path
```
Comment (rk3rn3r) [inline .github/workflows/integration.yml:55]: Do you use yarn? Else this can be removed.

File: .github/workflows/integration.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Cache local Yarn repository
```
Comment (rk3rn3r) [inline .github/workflows/integration.yml:59]: same here

**Comment by rk3rn3r** [comment]
As of Ashique's absence I will finish his work (some minor renaming and removing unused build steps) and merge the PR afterwards.

**Comment by rk3rn3r** [comment]
Merged. Thx @uidoyen for your hard work on this one!

**Pull request merged**